### PR TITLE
Link to GitHub source page (CuPy)

### PR DESCRIPTION
**Pull request by niboshi**
This PR is a port from https://github.com/chainer/chainer/pull/2758 ~(currently not merged yet)~.

### CODE REVIEW

**Comment by okuta** [comment]
I test link on [current RTD page](http://docs.cupy.chainer.org/en/latest/index.html). So, I can jump to github page.
What problem does this PR solve?

**Comment by niboshi** [comment]
I guess the link you mentioned is for .rst files. This PR aims to put GitHub links for each documented objects (classes and functions) to .py files.

**Comment by unnonouno** [comment]
Ah, did we release v1.0.0.1 on Github? URL is not found.

**Comment by niboshi** [comment]
Do you mean you can't follow the links from locally-built docs?
Yes, I have noticed the problem, but I think it's ok, because:
* For `latest` version in readthedocs, the links will be `master`, not `1.0.0.1`.
* For `stable` and other specific versions, at the moment these docs are built in readthedocs, the version should have been updated, to e.g. `1.0.1`. So in GitHub it should also have `1.0.1` tagged.

In summary I think it's the problem only of locally-build docs.

**Comment by unnonouno** [comment]
I see. OK, LGTM
